server/src/main/java/org/opensearch/rest/action/search/RestMultiSearchAction.java

```diff
@@ -166,6 +168,11 @@ public static MultiSearchRequest parseRequest(RestRequest restRequest,
             if (maxConcurrentShardRequests != null) {
                 request.setMaxConcurrentShardRequests(maxConcurrentShardRequests);
             }
+            // if cancel_after_timeinterval parameter is set at per search request level than that is used otherwise one set at
+            // multi search request level will be used
+            if (cancelAfterTime != null && request.getCancelAfterTimeInterval() == null) {
```

Comment:
I think that the first null check is redundant since you only care if `request.getCancelAfterTimeInterval() == null` and the `cancelAfterTimeInterval` can be either null or not. 
```suggestion
            if (request.getCancelAfterTimeInterval() == null) {
```